pkg/handler/tool_handler.go

```diff
@@ -93,9 +96,26 @@ func (h *ToolHandler) run(tool string, cmds []string) error {
 			return fmt.Errorf("failed to run command %q, error %w", c, err)
 		}
 		// Empty line in between commands.
-		if i < (len(cmds) - 1) {
+		if h.stdout != nil && i < (len(cmds)-1) {
 			fmt.Fprintf(cmd.Stdout, "\n")
 		}
 	}
 	return nil
 }
+
+func splitArgs(str string) []string {
```

Comment:
As we discussed offline, we can try run `bash -c "gcloud xxx"` instead. Like you said, that's likely safer